source/event_stream_rpc_server.c

```diff
@@ -1004,7 +1014,12 @@ static void s_route_message_by_type(
                 AWS_LS_EVENT_STREAM_RPC_SERVER, "id=%p: invoking on_incoming_stream callback", (void *)connection);
             if (connection->on_incoming_stream(
                     continuation->connection, continuation, operation_name, &options, connection->user_data)) {
-                aws_event_stream_rpc_server_continuation_release(continuation);
+
+                /*
+                 * We don't release the continuation here (unlike above in the put failure path) because the on remove
+                 * function for the continuation table (s_continuation_destroy) will invoke continuation release.
+                 */
+                aws_hash_table_remove(&connection->continuation_table, &continuation->stream_id, NULL, NULL);
```

Comment:
I'm a bit confused about the ref-counting.
at line 994 it gets initialized to 1
then at 1012 it gets bumped to 2
then removing it from the hash-table will put it back

is the user expected to release() the continuation in the case that they return AWS_OP_ERR?

please add documentation to `aws_event_stream_rpc_server_on_incoming_stream_fn()` regarding the ref-counting expectations